crates/apis/src/console/config.rs

```diff
@@ -0,0 +1,51 @@
+use std::io::{self, Write};
+
+use crate::APIConfig;
+
+/// A selection of possible destination streams for `console.log` and
+/// `console.error`.
+#[derive(Debug)]
+pub enum LogStream {
+    /// The standard output stream.
+    StdOut,
+    /// The standard error stream.
+    StdErr,
+}
+
+impl LogStream {
+    pub(super) fn to_stream(&self) -> Box<dyn Write + 'static> {
+        match self {
+            Self::StdErr => Box::new(io::stderr()),
+            Self::StdOut => Box::new(io::stdout()),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct ConsoleConfig {
+    pub(super) log_stream: LogStream,
+    pub(super) error_stream: LogStream,
+}
+
+impl Default for ConsoleConfig {
+    fn default() -> Self {
+        Self {
+            log_stream: LogStream::StdOut,
+            error_stream: LogStream::StdErr,
+        }
+    }
+}
+
+impl APIConfig {
+    /// Sets the destination stream for `console.log`.
+    pub fn log_stream(&mut self, stream: LogStream) -> &mut Self {
```

Comment:
What happens if another API happens to implement the same name `log_stream` here?

Reply:
Just to expand on my comment and concern here:

I'm wondering if this pattern would mean that API creators will have to be aware of the possible names of other API configs so they don't clobber. That might be a burden we'd want to avoid

Reply:
Right, that's good that there's a warning at least. But I assume this warning would also only appear if they have the conflicting feature turned on. IMO it feels like we'd want to avoid this issue altogether. What do you think?

Reply:
> I'm wondering if this pattern would mean that API creators will have to be aware of the possible names of other API configs so they don't clobber.

I don't think API creators would have to be aware of all possible names. The compiler will let them know if they clobber if all features are turned on. They can also look at the docs for the crate or some kind of autocompletion to get a full list of methods on `APIConfig` if they clobber and want to ensure the next name they try isn't already defined on `APIConfig`.

Reply:
For sure, I guess I don't like that they are limited by the names they can pick here, especially if these will end up being separate crates eventually.

If you feel that's not that big of a deal, I'm okay with approving this

Reply:
Just catching up on this. I do like the approach followed in this PR in order to ease the `cfg` complexity (e.g. having dedicated structs/enums for each API configuration).

On the other hand, my gut tells me that we could start simple, by following the second approach described in the previous comment

```rust
config.log_stream(...)
```


The reason for this preference is mostly to stick to the principle that configuration should happen via a single, top-level entry point, with the hope to make it easier to reason about. In the first approach we'd almost be configuring each API separately, which risks simplicity.

All that said, this is something that we can keep an eye on, and change if there's need to. I do agree though, that consumers will be limited by the names that they can pick for each API, but hopefully this won't happen often.

Reply:
I'm going to go ahead and merge this then since it uses the second approach. We do use sub-structs behind the scenes but the consumer facing API for `APIConfig` hides all of that.